src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/nocddr_fastaccess_hwemu.cxx

```diff
@@ -102,12 +102,20 @@ bool nocddr_fastaccess_hwemu::init(std::string filename, std::string simdir)
     ss1 >> foffset;
     ss2 >> fsize;
     fname = simdir + "/" + fname;
-    std::cerr << "Creating fname=" << fname << std::hex << ", offset=0x" << foffset << ", size=0x" << fsize << std::endl;
     //mmap the file create a tuple and add to the array
     int fd = open(fname.c_str(), (O_CREAT | O_RDWR), 0666);
-    int rf = ftruncate(fd, fsize);
-    if (rf == -1)
+    if(fd >= 0)
     {
+      int rf = ftruncate(fd, fsize);
+      if (rf == -1)
+      {
+        close(fd);
+        return false;
+      }
+    }
+    else
+    {
+      close(fd);
       return false;
```

Comment:
I assume a close() was missing before this change, which just goes to show that `open()` has no place in C++.  However, I am not sure of the truncate alternative with fstream, so let it be.  However the logic looks unnecessarily complicated, can't it be just:
```
if (fd > 0)
  ftruncate(fd, fsize);

close(fd);
return false;
```

Reply:
As we are using fd in very next line(121), so it should be open when ftruncate succeeds. As per your suggestion it will be closed in this case too, so keeping it open when ftruncate succeeds should be good, right?

Reply:
Got you.  You are right.

Reply:
All this useful information should be in comments